cmd/show/dashboard.go

```diff
@@ -0,0 +1,323 @@
+package show
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/charmbracelet/bubbles/table"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	client "github.com/open-sauced/go-api/client"
+)
+
+const (
+	newContributorsView = iota
+	alumniContributorsView
+)
+
+// DashboardModel holds all the information related to the repository queried (issues, stars, new contributors, alumni contributors)
+type DashboardModel struct {
+	newContributorsTable    table.Model
+	alumniContributorsTable table.Model
+	RepositoryInfo          *client.DbRepo
+	tableView               int
+}
+
+// SelectMsg: message to signal the main model that we want to go to the contributor model when 'enter' is pressed
+type SelectMsg struct {
+	contributorName string
+}
+
+// InitDashboard: initializes the dashboard model
+func InitDashboard(opts *Options) (DashboardModel, error) {
+	model := DashboardModel{}
+	err := validateShowQuery(opts)
+	if err != nil {
+		return model, err
+	}
+	ownerRepo := strings.Split(opts.RepoName, "/")
+
+	resp, r, err := opts.APIClient.RepositoryServiceAPI.FindOneByOwnerAndRepo(opts.ServerContext, ownerRepo[0], ownerRepo[1]).Execute()
+	if err != nil {
+		return model, err
+	}
+
+	if r.StatusCode != 200 {
+		return model, fmt.Errorf("HTTP status: %d", r.StatusCode)
+	}
+
+	model.RepositoryInfo = resp
+
+	var (
+		errorChan = make(chan error)
+		doneChan  = make(chan bool)
+		wg        sync.WaitGroup
+	)
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		newContributors, err := FetchNewContributors(opts, resp.Id)
+		if err != nil {
+			errorChan <- err
+			return
+		}
+		model.newContributorsTable = setupContributorsTable(newContributors)
+	}()
+
+	go func() {
+		defer wg.Done()
+		alumniContributors, err := FetchAlumniContributors(opts, resp.Id)
+		if err != nil {
+			errorChan <- err
+			return
+		}
+		model.alumniContributorsTable = setupContributorsTable(alumniContributors)
+	}()
+
+	go func() {
+		wg.Wait()
+		close(doneChan)
+	}()
+
+	select {
+	case <-doneChan:
+		break
+	case err := <-errorChan:
```

Comment:
I think I gave similar feedback to @cecobask (and I'm probably misremembering how these channels work) but won't this only take the first error and squash any other errors that land in this channel? We should surface any error that might happen.

Reply:
Yep, that is correct only the first error will be shown the rest is squashed, was not sure if we were going for the errgroup pattern for these type of concurrent operations, but makes sense to surface all errors.